core/arch/emit_utils_shared.c

```diff
@@ -5625,6 +5625,20 @@ emit_special_ibl_xfer(dcontext_t *dcontext, byte *pc, generated_code_t *code, ui
             dcontext, opnd_create_reg(SCRATCH_REG1),
             OPND_TLS_FIELD(get_ibl_entry_tls_offs(dcontext, ibl_linked_tgt))));
     APP(&ilist, XINST_CREATE_jump_reg(dcontext, opnd_create_reg(SCRATCH_REG1)));
+    /* Unlike X86 and ARM/AArch32 which use 1 instruction for an indirect jump,
+     * AArch64 requires 2 instructions: LDR+BR. In the case where cache line
+     * alignment places LDR+BR at the end of the IBL transfer stub, a relink
+     * (see relink_special_ibl_xfer) will result in the BR instruction
+     * overwriting the first instruction of the next stub, (currently
+     * clean_call_save). This code checks if LDR+BR are at the end and adds a
+     * NOP to trigger padding for cache line alignment when emitted.
```

Comment:
But this will still turn LDR;BR;NOP into LDR;LDR;BR which does not seem ideal to have an extra LDR to the same reg.  It seems cleaner to fix the root of the problem: point the special_ibl_unlink_offs at the LDR instead of the BR, right?  And have the relink routine only write the LDR since the BR does not need to change.

Reply:
A separate topic: relink_special_ibl_xfer is modifying a LDR instruction which is not on the list of opcodes that do not require synchronization with other threads when changing: so this code change method is technically unsafe.  We modified the aarch64 exit stubs to only change either data or opcodes on the list; probably we should do the same thing here.  Could we change a TLS value instead of changing the LDR to point to different TLS slots?  Currently this is under #1911.